### PR TITLE
frontend: support building in openEuler

### PR DESCRIPTION
**Pull request by pkking**
Some of the python packages have different name in openEuler, so i switch to `python3dist(xxx)`

### CODE REVIEW

File: frontend/copr-frontend.spec [hunk 1]
```diff
@@ -85,15 +85,18 @@ BuildRequires: python3-dateutil
 BuildRequires: python3-decorator
 BuildRequires: python3-flask
 BuildRequires: python3-templated-dictionary
-%if 0%{?fedora} >= 31
-BuildRequires: python3-flask-caching
+%if 0%{?fedora} >= 31 || 0%{?openEuler}
+BuildRequires: python3dist(flask-caching)
 %else
 BuildRequires: python3-flask-cache
 %endif
```
Comment (praiskup) [inline frontend/copr-frontend.spec:92]: Fedora 31 is not supported, please drop the %if-else condition.

File: frontend/copr-frontend.spec [hunk 2]
```diff
@@ -138,23 +138,28 @@ Recommends: fedora-messaging
...
-Requires: python3-CommonMark
 Requires: python3-alembic
 Requires: python3-blinker
 Requires: python3-copr-common >= %copr_common_version
 Requires: python3-dateutil
 Requires: python3-email-validator
 Requires: python3-flask
-%if 0%{?fedora} >= 31
-Requires: python3-flask-caching
+%if 0%{?fedora} >= 31 || 0%{?openEuler}
```
Comment (praiskup) [inline frontend/copr-frontend.spec:147]: Same here.

File: frontend/copr-frontend.spec [hunk 3]
```diff
@@ -138,23 +138,28 @@ Recommends: fedora-messaging
...
+Requires: python3dist(commonmark)
+Requires: python3dist(flask-openid)
+Requires: python3dist(flask-sqlalchemy)
+Requires: python3dist(sqlalchemy-utils)
+Requires: python3dist(pylibravatar)
+Requires: python3dist(wtforms) >= 2.2.1
+Requires: python3dist(flask-wtf)
+Requires: python3dist(xstatic-datatables)
+Requires: python3dist(xstatic-patternfly)
+Requires: python3dist(xstatic-bootstrap-scss)
```
Comment (praiskup) [inline frontend/copr-frontend.spec:161]: This is hard to detect at build time;  but ` xstatic-bootstrap-scss-common` do not provide `python3dist`, are you sure this works on openEuler?

**Comment by praiskup** [comment]
I'd vote for switching all the dependencies to the same format, not just some of those.  Team, WDYT?
Also, can you please keep the deps on the same line and just change the format?  Just to simplify the review (so we are sure nothing is lost, or added).

**Comment by Conan-Kudo** [comment]
You should be using `python3dist()` format for everything anyway. There's no reason not to...

File: frontend/copr-frontend.spec [hunk 4]
```diff
@@ -1,5 +1,5 @@
 %bcond_without check
-%if 0%{?fedora} < 32
+%if 0%{?fedora} < 32 && !0%{?openEuler}
```
Comment (praiskup) [inline frontend/copr-frontend.spec:2]: This condition can be dropped as well.  Just use %bcond_without unconditionally.

**Comment by praiskup** [comment]
We don't have the installability test :-( but I see this issue now:
```
  - nothing provides python3dist(mod_wsgi) needed by copr-frontend-1.200-1.git.37.11129e7.fc38.noarch from copr:copr.fedorainfracloud.org:group_copr:copr-pull-requests:pr:2784
  - nothing provides python3dist(zmq) needed by copr-frontend-1.200-1.git.37.11129e7.fc38.noarch from copr:copr.fedorainfracloud.org:group_copr:copr-pull-requests:pr:2784
```

**Comment by praiskup** [comment]
This looks good to me, can anyone else have yet another look?

**Review by FrostyX — APPROVED** [review]

**Pull request merged**